db/ens.go

```diff
@@ -0,0 +1,506 @@
+				continue
+			}
+
+			nameHash, err := go_ens.NameHash(nameRenewed.Name)
+			if err != nil {
+				utils.LogError(err, "error hashing ens name", 0)
+				continue
+			}
+			keys[fmt.Sprintf("%s:ENS:I:H:%x:%x", bigtable.chainId, nameHash, tx.GetHash())] = true
+			keys[fmt.Sprintf("%s:ENS:V:N:%s", bigtable.chainId, nameRenewed.Name)] = true
+
+		} else if foundNameChangedIndex > -1 && foundNewOwnerIndex > -1 { // we found a name change event
+
+			log := logs[foundNewOwnerIndex]
+			topics := make([]common.Hash, 0, len(log.GetTopics()))
+
+			for _, lTopic := range log.GetTopics() {
+				topics = append(topics, common.BytesToHash(lTopic))
+			}
+			newOwnerLog := eth_types.Log{
+				Address:     common.BytesToAddress(log.GetAddress()),
+				Data:        log.Data,
+				Topics:      topics,
+				BlockNumber: blk.GetNumber(),
+				TxHash:      common.BytesToHash(tx.GetHash()),
+				TxIndex:     uint(i),
+				BlockHash:   common.BytesToHash(blk.GetHash()),
+				Index:       uint(foundNewOwnerIndex),
+				Removed:     log.GetRemoved(),
+			}
+
+			newOwner, err := filterer.ParseNewOwner(newOwnerLog)
+			if err != nil {
+				utils.LogError(err, fmt.Errorf("indexing of new owner event failed parse event at index %v", foundNewOwnerIndex), 0)
+				continue
+			}
+
+			keys[fmt.Sprintf("%s:ENS:I:A:%x:%x", bigtable.chainId, newOwner.Owner, tx.GetHash())] = true
+			keys[fmt.Sprintf("%s:ENS:V:A:%x", bigtable.chainId, newOwner.Owner)] = true
+		}
+		// We found a change address event, there can be multiple within one transaction
+		for _, addressChangeIndex := range foundAddressChangedIndices {
+
+			log := logs[addressChangeIndex]
+			topics := make([]common.Hash, 0, len(log.GetTopics()))
+
+			for _, lTopic := range log.GetTopics() {
+				topics = append(topics, common.BytesToHash(lTopic))
+			}
+
+			addressChangedLog := eth_types.Log{
+				Address:     common.BytesToAddress(log.GetAddress()),
+				Data:        log.Data,
+				Topics:      topics,
+				BlockNumber: blk.GetNumber(),
+				TxHash:      common.BytesToHash(tx.GetHash()),
+				TxIndex:     uint(i),
+				BlockHash:   common.BytesToHash(blk.GetHash()),
+				Index:       uint(addressChangeIndex),
+				Removed:     log.GetRemoved(),
+			}
+
+			addressChanged, err := filterer.ParseAddressChanged(addressChangedLog)
+			if err != nil {
+				utils.LogError(err, "indexing of address change event failed parse event at index ", 0)
+				continue
+			}
+
+			keys[fmt.Sprintf("%s:ENS:I:H:%x:%x", bigtable.chainId, addressChanged.Node, tx.GetHash())] = true
+			keys[fmt.Sprintf("%s:ENS:V:H:%x", bigtable.chainId, addressChanged.Node)] = true
+
+		}
+	}
+	for key := range keys {
+		mut := gcp_bigtable.NewMutation()
+		mut.Set(DEFAULT_FAMILY, key, gcp_bigtable.Timestamp(0), nil)
+
+		bulkData.Keys = append(bulkData.Keys, key)
+		bulkData.Muts = append(bulkData.Muts, mut)
+	}
+
+	return bulkData, bulkMetadataUpdates, nil
+}
+
+func (bigtable *Bigtable) ImportEnsUpdates(client *ethclient.Client) error {
```

Comment:
This whole function retrieves one ENS name after the other from the node. This should, if possible, be done in batches. Please compare that with `GetBalances` here:
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/79e21c30410ee9f60172ab37cbd6fe25f0fc0822/rpc/erigon.go#L481

If this cannot be achieved easily, I suggest to discuss this topic with @peterbitfly. From my POV the node will by far not be queried as much as for balances so maybe the approach is still okay.

Reply:
batch batch batch - let it burn!

Reply:
As discussed, batching those calls does not seem possible.